### PR TITLE
Add launcher SBOM input

### PR DESCRIPTION
**Pull request by natalieparellano**
As a follow on to https://github.com/buildpacks/lifecycle/pull/944, this PR introduces a `-launcher-sbom` flag to the exporter & creator so that users who supply their own launcher can supply an SBOM for that launcher.

Related spec PR: https://github.com/buildpacks/spec/pull/332/files?diff=split&w=1

### CODE REVIEW

File: exporter_test.go [hunk 1]
```diff
@@ -1075,21 +1076,6 @@ version = "4.5.6"
...
-
-				extensions := exporter.SBOMExtensions()
-				components := exporter.SBOMComponents()
-
-				for _, component := range components {
-					for _, extension := range extensions {
-						assertLogEntry(t, logHandler, component+extension+" is missing")
-					}
-				}
-			})
```
Comment (natalieparellano) [inline exporter_test.go:1092]: Moved out of nested context (for readability)

**Review by jjbustamante — APPROVED** [review]
I loved how simply is adding a new flag after the refactoring! Great Job @natalieparellano

**Comment by natalieparellano** [comment]
Sigh... again I think Codecov is being too picky (the newly exported constants are being tested implicitly elsewhere). @jabrown85 do you have an opinion?

**Comment by jabrown85** [comment]
I'm fine with it!

**Pull request merged**